### PR TITLE
[Build System: CMake] Convert the SWIFT_STDLIB_BUILD_TYPE variable into a CMake cache variable in cmake/modules/StandaloneOverlay.cmake

### PR DESCRIPTION
**Pull request by Rostepher**

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please smoke test

**Comment by Rostepher** [comment]
@swift-ci please test Linux platform

**Comment by Rostepher** [comment]
@swift-ci please smoke test Linux platform

File: cmake/modules/StandaloneOverlay.cmake [hunk 1]
```diff
@@ -10,9 +10,13 @@ endif()
 
 set(CMAKE_INSTALL_PREFIX "${SWIFT_DEST_ROOT}${TOOLCHAIN_DIR}/usr")
 
+set(SWIFT_STDLIB_BUILD_TYPE "Release" CACHE STRING
+    "Build type for the Swift standard library and SDK overlays [Debug, RelWithDebInfo, Release, MinSizeRel]")
+set_property(CACHE SWIFT_STDLIB_BUILD_TYPE PROPERTY
```
Comment (compnerd) [inline cmake/modules/StandaloneOverlay.cmake]: Why the property?  The `set(...CACHE...)` should be sufficient.

Comment (Rostepher) [reply cmake/modules/StandaloneOverlay.cmake]: I'm just duplicating what's already in the top-level `CMakeLists.txt` file here.

**Comment by Rostepher** [comment]
@swift-ci please smoke test Linux platform

**Pull request merged**